test/integration/common/StudioActions.kt

```diff
@@ -45,39 +45,71 @@ import java.nio.file.Path
 import java.nio.file.Paths
 import kotlin.test.assertEquals
 import kotlin.test.assertTrue
-import kotlin.test.fail
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
 import mu.KotlinLogging
 
 object StudioActions {
     private val LOGGER = KotlinLogging.logger {}
 
-    suspend fun clickIcon(composeRule: ComposeContentTestRule, icon: Icon, delayMillis: Int = Delays.RECOMPOSE) {
-        clickText(composeRule, icon.unicode, delayMillis)
+    suspend fun clickIcon(composeRule: ComposeContentTestRule, icon: Icon) {
+        clickText(composeRule, icon.unicode)
     }
 
-    suspend fun clickText(composeRule: ComposeContentTestRule, text: String, delayMillis: Int = Delays.RECOMPOSE) {
+    suspend fun clickText(composeRule: ComposeContentTestRule, text: String) {
+        waitUntilNodeWithTextIsClickable(composeRule, text)
         composeRule.onNodeWithText(text).performClick()
-        delayAndRecompose(composeRule, delayMillis)
+        delayAndRecompose(composeRule)
     }
 
-    fun clickAllInstancesOfText(composeRule: ComposeContentTestRule, text: String) {
+    suspend fun clickAllInstancesOfText(composeRule: ComposeContentTestRule, text: String) {
         val length = composeRule.onAllNodesWithText(text).fetchSemanticsNodes().size
         for (i in 0 until length) {
             composeRule.onAllNodesWithText(text)[i].performClick()
         }
+        delayAndRecompose(composeRule)
     }
 
-    fun clickAllInstancesOfIcon(composeRule: ComposeContentTestRule, icon: Icon) {
+    suspend fun clickAllInstancesOfIcon(composeRule: ComposeContentTestRule, icon: Icon) {
         clickAllInstancesOfText(composeRule, icon.unicode)
     }
 
-    fun assertNodeExistsWithText(composeRule: ComposeContentTestRule, text: String): SemanticsNodeInteraction {
-        return composeRule.onNodeWithText(text).assertExists()
+    suspend fun waitUntilNodeWithTextIsClickable(composeRule: ComposeContentTestRule, text: String) {
+        waitUntilAssertionIsTrue(composeRule) {
+            assertionDoesNotThrowException {
+                composeRule.onNodeWithText(text).assertHasClickAction()
+            }
+        }
+    }
+
+    suspend fun waitUntilNodeWithTextExists(composeRule: ComposeContentTestRule, text: String) {
+        waitUntilAssertionIsTrue(composeRule) {
+            assertionDoesNotThrowException {
+                composeRule.onNodeWithText(text).assertExists()
+            }
+        }
+    }
+
+    private fun assertionDoesNotThrowException(assertion: () -> Any): Boolean {
+        return try {
+            assertion()
+            true
+        } catch (e: Exception) {
+            false
+        }
     }
```

Comment:
Some functions, such as `assertExists(errorMessageOnFail: String? = null): SemanticsNodeInteraction` from `androidx.compose.ui.test`, return 'SemanticsNodeInteraction' rather than a boolean on the function `assertExists`. Ideally they would return a true or false rather than the node itself or throwing an exception. This is not the case.

In order to deal with these logically, we use `assertionDoesNotThrowException` that treats any wrapped function that does not throw an exception as returning true, or returning false if it does.